Alloy/MTLSharedGraphicsBuffer.swift

```diff
@@ -0,0 +1,150 @@
+import Foundation
+import Accelerate
+import Metal
+
+@available(iOS 12.0, macOS 10.14, *)
+final public class MTLSharedGraphicsBuffer {
+
+    public enum Error: Swift.Error {
+        case initializationFailed
+        case unsupportedPixelFormat
+    }
+
+    public let texture: MTLTexture
+    public let pixelBuffer: CVPixelBuffer
+    public let buffer: MTLBuffer
+    private(set) public var vImageBuffer: vImage_Buffer
+    public let cgContext: CGContext
+    public let pixelFormat: MTLPixelFormat
+    public let cvPixelFormat: OSType
+    public let rect: CGRect
+    public let region: MTLRegion
+
+    private var allocationPointer: UnsafeMutableRawPointer!
+
+    public init(context: MTLContext,
+                width: Int,
+                height: Int,
+                pixelFormat: MTLPixelFormat,
+                colorSpace: CGColorSpace,
+                bitmapInfo: UInt32,
+                usage: MTLTextureUsage = [.shaderRead, .shaderWrite, .renderTarget]) throws {
+        guard let cvPixelFormat = pixelFormat.compatibleCVPixelFormat,
+              let pixelFormatSize = pixelFormat.size,
+              let bitsPerComponent = pixelFormat.bitsPerComponent
+        else { throw Error.unsupportedPixelFormat }
+
+        let textureDescriptor = MTLTextureDescriptor()
+        textureDescriptor.pixelFormat = pixelFormat
+        textureDescriptor.usage = usage
+        textureDescriptor.width = width
+        textureDescriptor.height = height
+        #if os(iOS) && !targetEnvironment(macCatalyst)
+        textureDescriptor.storageMode = .shared
+        #elseif os(macOS) || (os(iOS) && targetEnvironment(macCatalyst))
+        textureDescriptor.storageMode = .managed
+        #endif
+
+        // MARK: - Page align allocation pointer.
+
+        /// The size of heap texture created from MTLBuffer.
+        let heapTextureSizeAndAlign = context.heapTextureSizeAndAlign(descriptor: textureDescriptor)
+
+        /// Current system's RAM page size.
+        let pageSize = Int(getpagesize())
+
+        /// Page aligned texture size.
+        ///
+        /// Get page aligned texture size.
+        /// It might be more than raw texture size, but we'll alloccate memory in reserve.
```

Comment:
I think docs are a bit excessive here, let's just add a link to initial article instead